modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/TxStateStorage.java

```diff
@@ -0,0 +1,116 @@
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Storage for transaction meta, {@link TxMeta}.
+ */
+public interface TxStateStorage extends AutoCloseable {
+    /**
+     * Start the storage.
+     *
+     * @throws IgniteInternalException with {@link Transactions#TX_STATE_STORAGE_CREATE_ERR} error code in case when
+     *                                 creation of the storage has failed.
+     */
+    void start();
+
+    /**
+     * Whether the storage is started.
+     *
+     * @return {@code true} if the storage is started, {@code false} otherwise.
+     */
+    boolean isStarted();
+
+    /**
+     * Stop the storage.
+     */
+    void stop() throws Exception;
+
+    /**
+     * Get tx meta by tx id.
+     *
+     * @param txId Tx id.
+     * @return Tx meta.
+     * @throws IgniteInternalException with {@link Transactions#TX_STATE_STORAGE_ERR} error code in case when
+     *                                 the operation has failed.
+     */
+    TxMeta get(UUID txId);
+
+    /**
+     * Put the tx meta into the storage.
+     *
+     * @param txId Tx id.
+     * @param txMeta Tx meta.
+     * @throws IgniteInternalException with {@link Transactions#TX_STATE_STORAGE_ERR} error code in case when
+     *                                 the operation has failed.
+     */
+    void put(UUID txId, TxMeta txMeta);
+
+    /**
+     * Atomically change the tx meta in the storage.
+     *
+     * @param txId Tx id.
+     * @param txStateExpected Tx state that is expected to be in the storage.
+     * @param txMeta Tx meta.
+     * @return Whether the CAS operation is successful.
+     * @throws IgniteInternalException with {@link Transactions#TX_STATE_STORAGE_ERR} error code in case when
+     *                                 the operation has failed.
+     */
+    boolean compareAndSet(UUID txId, @NotNull TxState txStateExpected, @NotNull TxMeta txMeta);
+
+    /**
+     * Remove the tx meta from the storage.
+     *
+     * @param txId Tx id.
+     * @throws IgniteInternalException with {@link Transactions#TX_STATE_STORAGE_ERR} error code in case when
+     *                                 the operation has failed.
+     */
+    void remove(UUID txId);
+
+    /**
+     * Removes all data from the storage and frees all resources.
+     *
+     * @throws IgniteInternalException with {@link Transactions#TX_STATE_STORAGE_DESTROY_ERR} error code in case when
+     *                                 the operation has failed.
+     */
+    void destroy();
+
+    /**
+     * Create a snapshot of the storage's current state in the specified directory.
+     *
+     * @param snapshotPath Snapshot path.
+     * @return Future which completes when the snapshot operation is complete.
+     */
+    CompletableFuture<Void> snapshot(Path snapshotPath);
+
+    /**
+     * Restore a storage from the snapshot.
+     *
+     * @param snapshotPath Snapshot path.
+     */
+    void restoreSnapshot(Path snapshotPath);
```

Comment:
We will also need scan in order to retrieve all existing tx meta, e.g. on primary replica switch including node restart.

Reply:
added `scan`.